src/main.zig

```diff
@@ -2177,13 +2177,37 @@ fn buildOutputType(
                 const sdk = std.zig.system.darwin.getDarwinSDK(arena, target_info.target) orelse
                     break :outer false;
                 native_darwin_sdk = sdk;
+
                 try clang_argv.ensureUnusedCapacity(2);
                 clang_argv.appendAssumeCapacity("-isysroot");
                 clang_argv.appendAssumeCapacity(sdk.path);
+
+                // if we're on darwin and nix, add a sysroot relative include path
+                if (paths.is_nix) {
+                    try clang_argv.ensureUnusedCapacity(paths.nix_include_dirs.items.len * 2);
+                    clang_argv.appendAssumeCapacity("-iwithsysroot");
+                    clang_argv.appendAssumeCapacity("/usr/include");
+                }
+
                 break :outer true;
             } else break :outer false;
         } else false;
 
+        // NOTE: nix-darwin supplies paths that are absolute and not relative to any sysroot/sdk
+        // so even if an sdk is found (XCode installation), nix include/framework dirs must be added
+        // without relation
+        try clang_argv.ensureUnusedCapacity(paths.nix_include_dirs.items.len * 2);
+        for (paths.nix_include_dirs.items) |include_dir| {
+            clang_argv.appendAssumeCapacity("-isystem");
+            clang_argv.appendAssumeCapacity(include_dir);
+        }
+
+        try clang_argv.ensureUnusedCapacity(paths.nix_framework_dirs.items.len * 2);
+        for (paths.nix_include_dirs.items) |framework_dir| {
```

Comment:
This should be `paths.nix_framework_dirs.items`

Reply:
You're correct, will change